modules/ml/src/main/java/org/apache/ignite/ml/environment/LearningEnvironmentBuilder.java

```diff
@@ -17,109 +17,78 @@
 
 package org.apache.ignite.ml.environment;
 
+import java.util.Random;
 import org.apache.ignite.ml.environment.logging.MLLogger;
-import org.apache.ignite.ml.environment.logging.NoOpLogger;
-import org.apache.ignite.ml.environment.parallelism.DefaultParallelismStrategy;
-import org.apache.ignite.ml.environment.parallelism.NoParallelismStrategy;
 import org.apache.ignite.ml.environment.parallelism.ParallelismStrategy;
+import org.apache.ignite.ml.math.functions.IgniteFunction;
+import org.apache.ignite.ml.math.functions.IgniteSupplier;
 
-/**
- * Builder for LearningEnvironment.
- */
-public class LearningEnvironmentBuilder {
-    /** Parallelism strategy. */
-    private ParallelismStrategy parallelismStgy;
-    /** Logging factory. */
-    private MLLogger.Factory loggingFactory;
-
+public interface LearningEnvironmentBuilder {
```

Comment:
It should be `Serializable` because we send it to worker nodes.

Reply:
Now we have `IgniteFunction<partIdx, X>` fields instead of `X` in this class.